### ISSUE TITLE
Error when poling 'URIError: URI malformed     at decodeURIComponent (<anonymous>)'

### ISSUE THREAD

**Issue by cschildhorn** [SeydX/homebridge-fritz-platform#110]
Plugin is unable to connect to devices. Worked for a short time with 5.0.3 release, currently I only see this error messages when plugin tries to pull data.

Just as example:

```
[11/11/2020, 12:28:36] [FritzPlatform] AP-Wohnzimmer: An error occured during polling state (router)
[11/11/2020, 12:28:36] [FritzPlatform] URIError: URI malformed
    at decodeURIComponent (<anonymous>)
    at Service.<anonymous> (/usr/local/lib/node_modules/homebridge-fritz-platform/node_modules/@seydx/fritzbox/src/lib/service.ts:172:63)
    at Generator.next (<anonymous>)
    at /usr/local/lib/node_modules/homebridge-fritz-platform/node_modules/@seydx/fritzbox/dist/lib/service.js:8:71
    at new Promise (<anonymous>)
    at __awaiter (/usr/local/lib/node_modules/homebridge-fritz-platform/node_modules/@seydx/fritzbox/dist/lib/service.js:4:12)
    at Service.execAction (/usr/local/lib/node_modules/homebridge-fritz-platform/node_modules/@seydx/fritzbox/dist/lib/service.js:159:16)
    at Fritzbox.<anonymous> (/usr/local/lib/node_modules/homebridge-fritz-platform/node_modules/@seydx/fritzbox/src/lib/fritzbox.ts:243:20)
    at Generator.next (<anonymous>)
    at fulfilled (/usr/local/lib/node_modules/homebridge-fritz-platform/node_modules/@seydx/fritzbox/dist/lib/fritzbox.js:5:58)
[11/11/2020, 12:28:36] [FritzPlatform] Hosts: An error occured during polling state (hosts)
[11/11/2020, 12:28:36] [FritzPlatform] URIError: URI malformed
    at decodeURIComponent (<anonymous>)
    at Service.<anonymous> (/usr/local/lib/node_modules/homebridge-fritz-platform/node_modules/@seydx/fritzbox/src/lib/service.ts:172:63)
    at Generator.next (<anonymous>)
    at /usr/local/lib/node_modules/homebridge-fritz-platform/node_modules/@seydx/fritzbox/dist/lib/service.js:8:71
    at new Promise (<anonymous>)
    at __awaiter (/usr/local/lib/node_modules/homebridge-fritz-platform/node_modules/@seydx/fritzbox/dist/lib/service.js:4:12)
    at Service.execAction (/usr/local/lib/node_modules/homebridge-fritz-platform/node_modules/@seydx/fritzbox/dist/lib/service.js:159:16)
    at Fritzbox.<anonymous> (/usr/local/lib/node_modules/homebridge-fritz-platform/node_modules/@seydx/fritzbox/src/lib/fritzbox.ts:243:20)
    at Generator.next (<anonymous>)
    at fulfilled (/usr/local/lib/node_modules/homebridge-fritz-platform/node_modules/@seydx/fritzbox/dist/lib/fritzbox.js:5:58)
[11/11/2020, 12:28:36] [FritzPlatform] Smarthome: An error occured during polling state (smarthome)
[11/11/2020, 12:28:36] [FritzPlatform] URIError: URI malformed
    at decodeURIComponent (<anonymous>)
    at Service.<anonymous> (/usr/local/lib/node_modules/homebridge-fritz-platform/node_modules/@seydx/fritzbox/src/lib/service.ts:172:63)
    at Generator.next (<anonymous>)
    at /usr/local/lib/node_modules/homebridge-fritz-platform/node_modules/@seydx/fritzbox/dist/lib/service.js:8:71
    at new Promise (<anonymous>)
    at __awaiter (/usr/local/lib/node_modules/homebridge-fritz-platform/node_modules/@seydx/fritzbox/dist/lib/service.js:4:12)
    at Service.execAction (/usr/local/lib/node_modules/homebridge-fritz-platform/node_modules/@seydx/fritzbox/dist/lib/service.js:159:16)
    at Fritzbox.<anonymous> (/usr/local/lib/node_modules/homebridge-fritz-platform/node_modules/@seydx/fritzbox/src/lib/fritzbox.ts:243:20)
    at Generator.next (<anonymous>)
    at fulfilled (/usr/local/lib/node_modules/homebridge-fritz-platform/node_modules/@seydx/fritzbox/dist/lib/fritzbox.js:5:58)
[11/11/2020, 12:28:46] [FritzPlatform] FRITZ!Box 6890 LTE: An error occured during polling state (router)
[11/11/2020, 12:28:46] [FritzPlatform] URIError: URI malformed
    at decodeURIComponent (<anonymous>)
    at Service.<anonymous> (/usr/local/lib/node_modules/homebridge-fritz-platform/node_modules/@seydx/fritzbox/src/lib/service.ts:172:63)
    at Generator.next (<anonymous>)
    at /usr/local/lib/node_modules/homebridge-fritz-platform/node_modules/@seydx/fritzbox/dist/lib/service.js:8:71
    at new Promise (<anonymous>)
    at __awaiter (/usr/local/lib/node_modules/homebridge-fritz-platform/node_modules/@seydx/fritzbox/dist/lib/service.js:4:12)
    at Service.execAction (/usr/local/lib/node_modules/homebridge-fritz-platform/node_modules/@seydx/fritzbox/dist/lib/service.js:159:16)
    at Fritzbox.<anonymous> (/usr/local/lib/node_modules/homebridge-fritz-platform/node_modules/@seydx/fritzbox/src/lib/fritzbox.ts:243:20)
    at Generator.next (<anonymous>)
    at fulfilled (/usr/local/lib/node_modules/homebridge-fritz-platform/node_modules/@seydx/fritzbox/dist/lib/fritzbox.js:5:58)
    at runNextTicks (internal/process/task_queues.js:58:5)
    at processTimers (internal/timers.js:494:9)
[11/11/2020, 12:28:46] [FritzPlatform] AP-Obergeschoss: An error occured during polling state (router)
[11/11/2020, 12:28:46] [FritzPlatform] URIError: URI malformed
    at decodeURIComponent (<anonymous>)
    at Service.<anonymous> (/usr/local/lib/node_modules/homebridge-fritz-platform/node_modules/@seydx/fritzbox/src/lib/service.ts:172:63)
    at Generator.next (<anonymous>)
    at /usr/local/lib/node_modules/homebridge-fritz-platform/node_modules/@seydx/fritzbox/dist/lib/service.js:8:71
    at new Promise (<anonymous>)
    at __awaiter (/usr/local/lib/node_modules/homebridge-fritz-platform/node_modules/@seydx/fritzbox/dist/lib/service.js:4:12)
    at Service.execAction (/usr/local/lib/node_modules/homebridge-fritz-platform/node_modules/@seydx/fritzbox/dist/lib/service.js:159:16)
    at Fritzbox.<anonymous> (/usr/local/lib/node_modules/homebridge-fritz-platform/node_modules/@seydx/fritzbox/src/lib/fritzbox.ts:243:20)
    at Generator.next (<anonymous>)
    at fulfilled (/usr/local/lib/node_modules/homebridge-fritz-platform/node_modules/@seydx/fritzbox/dist/lib/fritzbox.js:5:58)
    at runNextTicks (internal/process/task_queues.js:58:5)
    at processTimers (internal/timers.js:494:9)
[11/11/2020, 12:28:46] [FritzPlatform] AP-Keller: An error occured during polling state (router)
[11/11/2020, 12:28:46] [FritzPlatform] URIError: URI malformed
    at decodeURIComponent (<anonymous>)
    at Service.<anonymous> (/usr/local/lib/node_modules/homebridge-fritz-platform/node_modules/@seydx/fritzbox/src/lib/service.ts:172:63)
    at Generator.next (<anonymous>)
    at /usr/local/lib/node_modules/homebridge-fritz-platform/node_modules/@seydx/fritzbox/dist/lib/service.js:8:71
    at new Promise (<anonymous>)
    at __awaiter (/usr/local/lib/node_modules/homebridge-fritz-platform/node_modules/@seydx/fritzbox/dist/lib/service.js:4:12)
    at Service.execAction (/usr/local/lib/node_modules/homebridge-fritz-platform/node_modules/@seydx/fritzbox/dist/lib/service.js:159:16)
    at Fritzbox.<anonymous> (/usr/local/lib/node_modules/homebridge-fritz-platform/node_modules/@seydx/fritzbox/src/lib/fritzbox.ts:243:20)
    at Generator.next (<anonymous>)
    at fulfilled (/usr/local/lib/node_modules/homebridge-fritz-platform/node_modules/@seydx/fritzbox/dist/lib/fritzbox.js:5:58)
    at runNextTicks (internal/process/task_queues.js:58:5)
    at processTimers (internal/timers.js:494:9)
[11/11/2020, 12:28:46] [FritzPlatform] AP-Schlafzimmer: An error occured during polling state (router)
[11/11/2020, 12:28:46] [FritzPlatform] URIError: URI malformed
    at decodeURIComponent (<anonymous>)
    at Service.<anonymous> (/usr/local/lib/node_modules/homebridge-fritz-platform/node_modules/@seydx/fritzbox/src/lib/service.ts:172:63)
    at Generator.next (<anonymous>)
    at /usr/local/lib/node_modules/homebridge-fritz-platform/node_modules/@seydx/fritzbox/dist/lib/service.js:8:71
    at new Promise (<anonymous>)
    at __awaiter (/usr/local/lib/node_modules/homebridge-fritz-platform/node_modules/@seydx/fritzbox/dist/lib/service.js:4:12)
    at Service.execAction (/usr/local/lib/node_modules/homebridge-fritz-platform/node_modules/@seydx/fritzbox/dist/lib/service.js:159:16)
    at Fritzbox.<anonymous> (/usr/local/lib/node_modules/homebridge-fritz-platform/node_modules/@seydx/fritzbox/src/lib/fritzbox.ts:243:20)
    at Generator.next (<anonymous>)
    at fulfilled (/usr/local/lib/node_modules/homebridge-fritz-platform/node_modules/@seydx/fritzbox/dist/lib/fritzbox.js:5:58)
    at runNextTicks (internal/process/task_queues.js:58:5)
    at processTimers (internal/timers.js:494:9)
[11/11/2020, 12:28:46] [FritzPlatform] AP-Wohnzimmer: An error occured during polling state (router)
[11/11/2020, 12:28:47] [FritzPlatform] URIError: URI malformed
    at decodeURIComponent (<anonymous>)
    at Service.<anonymous> (/usr/local/lib/node_modules/homebridge-fritz-platform/node_modules/@seydx/fritzbox/src/lib/service.ts:172:63)
    at Generator.next (<anonymous>)
    at /usr/local/lib/node_modules/homebridge-fritz-platform/node_modules/@seydx/fritzbox/dist/lib/service.js:8:71
    at new Promise (<anonymous>)
    at __awaiter (/usr/local/lib/node_modules/homebridge-fritz-platform/node_modules/@seydx/fritzbox/dist/lib/service.js:4:12)
    at Service.execAction (/usr/local/lib/node_modules/homebridge-fritz-platform/node_modules/@seydx/fritzbox/dist/lib/service.js:159:16)
    at Fritzbox.<anonymous> (/usr/local/lib/node_modules/homebridge-fritz-platform/node_modules/@seydx/fritzbox/src/lib/fritzbox.ts:243:20)
    at Generator.next (<anonymous>)
    at fulfilled (/usr/local/lib/node_modules/homebridge-fritz-platform/node_modules/@seydx/fritzbox/dist/lib/fritzbox.js:5:58)
    at runNextTicks (internal/process/task_queues.js:58:5)
    at processTimers (internal/timers.js:494:9)
[11/11/2020, 12:28:47] [FritzPlatform] Hosts: An error occured during polling state (hosts)
[11/11/2020, 12:28:47] [FritzPlatform] URIError: URI malformed
    at decodeURIComponent (<anonymous>)
    at Service.<anonymous> (/usr/local/lib/node_modules/homebridge-fritz-platform/node_modules/@seydx/fritzbox/src/lib/service.ts:172:63)
    at Generator.next (<anonymous>)
    at /usr/local/lib/node_modules/homebridge-fritz-platform/node_modules/@seydx/fritzbox/dist/lib/service.js:8:71
    at new Promise (<anonymous>)
    at __awaiter (/usr/local/lib/node_modules/homebridge-fritz-platform/node_modules/@seydx/fritzbox/dist/lib/service.js:4:12)
    at Service.execAction (/usr/local/lib/node_modules/homebridge-fritz-platform/node_modules/@seydx/fritzbox/dist/lib/service.js:159:16)
    at Fritzbox.<anonymous> (/usr/local/lib/node_modules/homebridge-fritz-platform/node_modules/@seydx/fritzbox/src/lib/fritzbox.ts:24
```

![9E12C078-F335-4D2C-B8D4-23F41EB73892](https://user-images.githubusercontent.com/516532/98806248-2d6f8e00-2419-11eb-8dca-dbc507d59b3f.jpeg)

**Comment by SeydX** [comment]
@cschildhorn 

i think u re using it without a password or?

**Comment by cschildhorn** [comment]
no, I configured Fritzbox with username + password in entered it in the config of the plugin

**Comment by SeydX** [comment]
@cschildhorn 

Something went wrong with your password. Try a different password

**Comment by kantnschleifr** [comment]
Receiving the same error message after upgrading from 5.0.3 to the latest version.
Users name + password had been untouched and still in config like before.

**Comment by cschildhorn** [comment]
I am still investigating this. I created a separate user for the plugin now with a super easy password. This works for the master. The repeaters can only have a FritzRepeater password or no-password.

If I now remove the username from the config of the repeaters I get this error:

`[11/11/2020, 14:20:34] [FritzPlatform] AP-Obergeschoss: There is no username configured for this router. This router will be skipped.`

If I now use the same username as for the master I get this error:
```
[11/11/2020, 14:30:44] [FritzPlatform] AP-Schlafzimmer: An error occured during polling state (router)
[11/11/2020, 14:30:44] [FritzPlatform] Error: Invalid response: 401:Unauthorized
    at Request._callback (/usr/local/lib/node_modules/homebridge-fritz-platform/node_modules/@seydx/fritzbox/src/lib/request.ts:30:13)
    at Request.self.callback (/usr/local/lib/node_modules/homebridge-fritz-platform/node_modules/request/request.js:185:22)
    at Request.emit (events.js:315:20)
    at Request.<anonymous> (/usr/local/lib/node_modules/homebridge-fritz-platform/node_modules/request/request.js:1154:10)
    at Request.emit (events.js:315:20)
    at IncomingMessage.<anonymous> (/usr/local/lib/node_modules/homebridge-fritz-platform/node_modules/request/request.js:1076:12)
    at Object.onceWrapper (events.js:421:28)
    at IncomingMessage.emit (events.js:327:22)
    at endReadableNT (_stream_readable.js:1327:12)
    at processTicksAndRejections (internal/process/task_queues.js:80:21)
 
```
Currently I am not able to find a way to activate the same username/password combination on the repeaters.

**Comment by cschildhorn** [comment]
Ok. Got it working.

I now changed the repeater password and removed all special characters (in my case it was a %). The username must be non-empty but is skipped and the password special character seams like properly escaped.

In homebridge-config-ui-x configuration the username has an 'required' flag which seams wrong because it must be filled but is not used.
![4E4B2D91-2E72-4E4E-AC64-A7E82188CE39](https://user-images.githubusercontent.com/516532/98820608-a7f6d880-242e-11eb-8029-86fcd75366c7.jpeg)

**Comment by kantnschleifr** [comment]
Again same here... after removing % which was contained in the password it works again. 
Many thanks.

**Comment by cschildhorn** [comment]
I think escaping would be possible for the password, right @SeydX ?

**Comment by SeydX** [comment]
Fixed in v5.0.7
